pkg/nfd-master/nfd-master.go

```diff
@@ -270,14 +291,16 @@ func (s *labelerServer) SetLabels(c context.Context, r *pb.SetLabelsRequest) (*p
 	if !s.args.NoPublish {
 		// Advertise NFD worker version and label names as annotations
 		keys := make([]string, 0, len(labels))
-		for k, _ := range labels {
+		for k := range labels {
 			keys = append(keys, k)
 		}
 		sort.Strings(keys)
 		annotations := Annotations{"worker.version": r.NfdVersion,
-			"feature-labels": strings.Join(keys, ",")}
+			"feature-labels":     strings.Join(keys, ","),
+			"extended-resources": strings.Join(s.args.ResourceLabels, ",")}
 
-		err := updateNodeFeatures(s.apiHelper, r.NodeName, labels, annotations)
+		resourceLabels := resourceLabels(s.args.ResourceLabels)
```

Comment:
I'd construct `resourceLabels` (or `extendedResources`) earlier, before the annotations. Only adding labels that exist (not everything that was listed on command line)